ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -803,7 +808,29 @@ public static boolean isPrivate(Map<String, Object> args) throws RestServiceExce
    * @throws RestServiceException if exception occurs during parsing the arg.
    */
   public static boolean isChunkUpload(Map<String, Object> args) throws RestServiceException {
-    return getBooleanHeader(args, Headers.CHUNK_UPLOAD, false);
+    return getBooleanHeader(args, Headers.CHUNK_UPLOAD, false, false);
+  }
+
+  /**
+   * Determine if {@link Headers#NAMED_UPSERT} is set in the request args.
+   * @param args The request arguments.
+   * @return {@code false} if {@link Headers#NAMED_UPSERT} is set to false else return true.
+   * @throws RestServiceException if exception occurs during parsing the arg.
+   */
+  public static boolean isUpsertForNamedBlob(final Map<String, Object> args) throws RestServiceException {
+    return getBooleanHeader(args, Headers.NAMED_UPSERT, false, true);
+  }
```

Comment:
MAke a new method to avoid this code cascade

```
  public static boolean getOptionalBooleanHeader(Map<String, Object> args, String header, boolean defaultVal)
      throws RestServiceException {
    String stringValue = getHeader(args, header, false);
    if (stringValue == null) {
      return defaultVal;
    } else if ("false".equalsIgnoreCase(stringValue)) {
      return false;
    } else if ("true".equalsIgnoreCase(stringValue)) {
      return true;
    } else {
      throw new RestServiceException(
          header + "[" + stringValue + "] has an invalid value (allowed values: true, false)",
          RestServiceErrorCode.InvalidArgs);
    }
  }
```

Reply:
I think my current implementation is better due to:
1. Better to be explicit: the previous way assume the defaultVal is False (when the header is not set). My current way make it explicit for that. And allow user to set defaultVal to true when needed (for upsert case).
2. Avoid code/logic duplication: With your proposed new method, there will be duplicate code.